### ISSUE TITLE
Card sorting information in a deck breaks Cube from Deck

### ISSUE THREAD

**Issue by Simown** [magefree/mage#2425]
I was testing the editor by making a Theros block cube. At the bottom of the .dck file there is the metadata used to sort the deck in the editor:

> LAYOUT MAIN:(1,6)(COLOR,false)|([THS:214],[THS:210],[THS:213],[THS:212],[THS:216],[JOU:159],[BNG:162],[JOU:164],[JOU:163],[JOU:165],[THS:227],[THS:228],[THS:225],[THS:223],[THS:224],[THS:226],[BNG:165],[BNG:163],[BNG:164])([THS:75],[THS:76],[BNG:60],[BNG:61],[JOU:61],[THS:80],[JOU:62],[JOU:63],[JOU:66],[BNG:65],[BNG:66],[THS:86],[JOU:68],[BNG:68],[BNG:69],[BNG:70],[THS:87],[JOU:70],[JOU:72],[THS:89],[JOU:73],[BNG:75],[THS:90],[THS:91],[THS:93],[JOU:74],[THS:96],[JOU:75],[THS:97],[THS:98],[THS:99],[JOU:78],[THS:100],[THS:101],[JOU:84],[THS:107],[THS:109],[BNG:86],[THS:110])([BNG:31],[BNG:32],[THS:40],[JOU:31],[THS:41],[THS:42],[THS:43],[JOU:32],[JOU:34],[THS:46],[JOU:35],[JOU:36],[JOU:37],[THS:47],[BNG:36],[BNG:39],[BNG:41],[JOU:38],[THS:50],[JOU:42],[BNG:45],[THS:57],[THS:58],[BNG:48],[JOU:45],[JOU:46],[THS:59],[THS:60],[BNG:49],[JOU:48],[JOU:51],[JOU:52],[BNG:52],[THS:64],[THS:65],[THS:68],[THS:72],[THS:73],[JOU:58])([THS:149],[THS:150],[THS:151],[THS:152],[THS:153],[BNG:118],[BNG:119],[THS:156],[JOU:121],[JOU:122],[THS:157],[THS:158],[JOU:124],[BNG:123],[JOU:126],[THS:159],[JOU:128],[BNG:125],[THS:161],[JOU:130],[THS:162],[BNG:127],[THS:163],[THS:164],[BNG:129],[THS:168],[JOU:133],[THS:170],[THS:172],[JOU:136],[JOU:137],[BNG:137],[THS:177],[THS:178],[THS:179],[THS:180],[BNG:143],[THS:183],[THS:184])([BNG:87],[THS:111],[THS:112],[BNG:88],[JOU:89],[BNG:89],[THS:116],[THS:118],[JOU:93],[THS:119],[JOU:94],[BNG:94],[THS:122],[BNG:97],[THS:124],[JOU:100],[THS:126],[THS:127],[THS:128],[JOU:103],[JOU:104],[BNG:102],[THS:131],[JOU:106],[BNG:106],[BNG:107],[JOU:108],[BNG:108],[JOU:110],[BNG:109],[BNG:110],[BNG:111],[THS:141],[THS:143],[THS:145],[JOU:115],[JOU:115],[THS:146],[JOU:116])([JOU:1],[JOU:3],[BNG:3],[BNG:4],[JOU:5],[BNG:5],[THS:3],[THS:6],[BNG:6],[JOU:6],[THS:7],[THS:8],[BNG:7],[JOU:10],[BNG:10],[THS:12],[THS:13],[BNG:12],[BNG:13],[BNG:14],[THS:16],[BNG:16],[JOU:13],[THS:18],[BNG:17],[BNG:18],[THS:19],[THS:20],[THS:22],[JOU:18],[JOU:19],[THS:25],[BNG:22],[BNG:23],[THS:26],[JOU:22],[JOU:23],[BNG:26],[BNG:28])
LAYOUT SIDEBOARD:(0,0)(CMC,false)|

This is fine and doesn't change the contents of my deck but because of this I am unable to import and use this deck as a Booster Draft Elimination (Cube), pressing "OK" causes this exception:

> java.lang.reflect.UndeclaredThrowableException
	at com.sun.proxy.$Proxy8.createTournamentTable(Unknown Source)
	at mage.remote.SessionImpl.createTournamentTable(SessionImpl.java:1044)
	at mage.client.SessionHandler.createTournamentTable(SessionHandler.java:230)
	at mage.client.dialog.NewTournamentDialog.btnOkActionPerformed(NewTournamentDialog.java:639)
	at mage.client.dialog.NewTournamentDialog.access$800(NewTournamentDialog.java:78)
	at mage.client.dialog.NewTournamentDialog$9.actionPerformed(NewTournamentDialog.java:353)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.Component.processMouseEvent(Component.java:6533)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at mage.client.dialog.MageDialog.startModal(MageDialog.java:141)
	at mage.client.dialog.MageDialog.show(MageDialog.java:82)
	at java.awt.Component.show(Component.java:1671)
	at java.awt.Component.setVisible(Component.java:1623)
	at javax.swing.JComponent.setVisible(JComponent.java:2644)
	at mage.client.dialog.MageDialog.setVisible(MageDialog.java:88)
	at mage.client.dialog.NewTournamentDialog.showDialog(NewTournamentDialog.java:134)
	at mage.client.table.TablesPanel.btnNewTournamentActionPerformed(TablesPanel.java:1263)
	at mage.client.table.TablesPanel.access$800(TablesPanel.java:107)
	at mage.client.table.TablesPanel$5.actionPerformed(TablesPanel.java:763)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:289)
	at java.awt.Component.processMouseEvent(Component.java:6533)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6298)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Caused by: org.jboss.remoting.InvocationFailureException: Unable to perform invocation; nested exception is: 
	java.io.EOFException: end of file
	at org.jboss.remoting.transport.socket.SocketClientInvoker.handleException(SocketClientInvoker.java:146)
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.transport(MicroSocketClientInvoker.java:981)
	at org.jboss.remoting.transport.bisocket.BisocketClientInvoker.transport(BisocketClientInvoker.java:470)
	at org.jboss.remoting.MicroRemoteClientInvoker.invoke(MicroRemoteClientInvoker.java:169)
	at org.jboss.remoting.Client.invoke(Client.java:2084)
	at org.jboss.remoting.Client.invoke(Client.java:879)
	at org.jboss.remoting.transporter.TransporterClient.invoke(TransporterClient.java:475)
	... 71 more
Caused by: java.io.EOFException: end of file
	at org.jboss.remoting.transport.socket.MicroSocketClientInvoker.transport(MicroSocketClientInvoker.java:928)
	... 76 more

Ultimately looks like it's having problems parsing the file. If I take the metadata out of the file I am able to start a draft. I didn't actually try constructed but it may have similar issues.

**Comment by spjspj** [comment]
From memory I was calling a util function to load in the cards from the deck.  This may need to be updated with the recent change to the deck editor

**Comment by spjspj** [comment]
``` 
Deck cubeFromDeck = new Deck();                   
                    try {
                        cubeFromDeck = Deck.load(DeckImporterUtil.importDeck(cubeFromDeckFilename), true, true);
                 
```

**Comment by spjspj** [comment]
Fixed with #2434 ?
